core/src/services/hdfs/writer.rs

```diff
@@ -44,18 +78,42 @@ impl oio::Write for HdfsWriter<hdrs::AsyncFile> {
             .map_err(new_std_io_error)
     }
 
+    fn poll_close(&mut self, cx: &mut Context<'_>) -> Poll<Result<()>> {
+        loop {
+            if let Some(mut fut) = self.fut.take() {
```

Comment:
Please do not attempt to `take` this feature directly, as it may not be completed in a single iteration. Use `as_mut()` and set to `None` once it's returns `Ready`.